### DISCUSSION TITLE
Test vectors

### DISCUSSION THREAD

**Discussion by cryptoquick** [LNP-BP/LNPBPs#131] [General]
It would really help in development to have a set of test vectors to verify certain activities such as minting, transfers, secondary transfers, and other more complex scenarios, especially as functionality is expanded. This way improvements can be made without compromising fundamental functionality. It would also help test certain assumptions made during development, and speed up review.

If something like this desired, where should these test vectors be made, and how should they work? Ideally the nodes and their methods could be tested in an ephemeral context, without necessarily needing to make shell commands.

To make this work easier, maybe embeddable API equivalents to the cli commands that make calls to microservices over zmq could be made available as a single, unified crate. This would make it easier to make calls to test functionality within Rust tests.

What do people think? Would this work be useful or necessary?

**Comment by dr-orlovsky** [comment]
I think this is one of the tasks of the primary importance

**Comment by dr-orlovsky** [comment]
@cryptoquick I think this should be transformed into an issue in RGB Node repository. Can you pls post it there?

**Reply by cryptoquick** [reply]
I was able to create [this issue](https://github.com/LNP-BP/LNPBPs/issues/150) from the discussion, maybe you can transfer it to the RGB Node repo? I'm not sure I have permission.

**Reply by dr-orlovsky** [reply]
Nope, GitHub doesn't allow transferring across the orgs. I will check what is with your permissions - anyone should be able to open issues there, especially you :)

Checked, the issues are open and you are the member of the org with all the right. Can you re-confirm that you are unable to create an issue there?
